tests/system/python/packages/test_pi_webapi.py

```diff
@@ -15,50 +15,88 @@
 
 import subprocess
 import http.client
-import json
 import pytest
 import os
 import time
 import utils
-
+from pathlib import Path
+import urllib.parse
 
 TEMPLATE_NAME = "template.json"
 ASSET = "FOGL-2964-e2e-CoAP"
 DATAPOINT = "sensor"
 DATAPOINT_VALUE = 20
+north_service_name = "NorthReadingsToPI_WebAPI"
 
-
-def get_ping_status(fledge_url):
-    _connection = http.client.HTTPConnection(fledge_url)
-    _connection.request("GET", '/fledge/ping')
-    r = _connection.getresponse()
-    assert 200 == r.status
-    r = r.read().decode()
-    jdoc = json.loads(r)
-    return jdoc
-
-
-def get_statistics_map(fledge_url):
-    _connection = http.client.HTTPConnection(fledge_url)
-    _connection.request("GET", '/fledge/statistics')
-    r = _connection.getresponse()
-    assert 200 == r.status
-    r = r.read().decode()
-    jdoc = json.loads(r)
-    return utils.serialize_stats_map(jdoc)
+# This  gives the path of directory where fledge is cloned. test_file < packages < python < system < tests < ROOT
+PROJECT_ROOT = Path(__file__).parent.parent.parent.parent.parent
+SCRIPTS_DIR_ROOT = "{}/tests/system/python/scripts/package/".format(PROJECT_ROOT)
 
 
-def get_asset_tracking_details(fledge_url, event=None):
-    _connection = http.client.HTTPConnection(fledge_url)
-    uri = '/fledge/track'
-    if event:
-        uri += '?event={}'.format(event)
-    _connection.request("GET", uri)
-    r = _connection.getresponse()
-    assert 200 == r.status
-    r = r.read().decode()
-    jdoc = json.loads(r)
-    return jdoc
+@pytest.fixture
+def reset_fledge(wait_time):
```

Comment:
don't we have this in conftest?

Reply:
The reset fixture in conftest is source code based and as this is a package-based test, it could not be used.